pkg/routing/table.go

```diff
@@ -1,135 +1,211 @@
 package routing
 
 import (
-	"bytes"
-	"encoding/json"
-	"fmt"
-	"strings"
+	"context"
+	"net/http"
 	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	httpv1alpha1 "github.com/kedacore/http-add-on/operator/apis/http/v1alpha1"
+	"github.com/kedacore/http-add-on/operator/generated/informers/externalversions"
+	informershttpv1alpha1 "github.com/kedacore/http-add-on/operator/generated/informers/externalversions/http/v1alpha1"
+	"github.com/kedacore/http-add-on/pkg/k8s"
+	"github.com/kedacore/http-add-on/pkg/util"
+)
+
+var (
+	errUnknownSharedIndexInformer = errors.New("informer is not cache.sharedIndexInformer")
+	errStartedSharedIndexInformer = errors.New("sharedIndexInformer has started, run more than once is not allowed")
+	errStoppedSharedIndexInformer = errors.New("sharedIndexInformer has stopped")
+	errNotSyncedTable             = errors.New("table has not synced")
 )
 
-type TableReader interface {
-	Lookup(string) (*Target, error)
-	Hosts() []string
-	HasHost(string) bool
+type Table interface {
+	util.HealthChecker
+
+	Start(ctx context.Context) error
+	Route(req *http.Request) *httpv1alpha1.HTTPScaledObject
+	HasSynced() bool
 }
-type Table struct {
-	fmt.Stringer
-	m map[string]Target
-	l *sync.RWMutex
+
+type table struct {
+	httpScaledObjectInformer                 sharedIndexInformer
+	httpScaledObjectEventHandlerRegistration cache.ResourceEventHandlerRegistration
+	httpScaledObjects                        map[types.NamespacedName]*httpv1alpha1.HTTPScaledObject
```

Comment:
This map use a "complex" (types.NamespacedName has 2 strings inside). 
Does it make sense if we generate the string concatenating both from our side and using `map[string]*httpv1alpha1.HTTPScaledObject`?